controllers/choose_removals.go

```diff
@@ -42,33 +42,67 @@ func (c ChooseRemovals) Reconcile(r *FoundationDBClusterReconciler, context ctx.
 		}
 	}
 
-	currentCounts := fdbtypes.CreateProcessCountsFromProcessGroupStatus(cluster.Status.ProcessGroups).Map()
+	currentCounts := fdbtypes.CreateProcessCountsFromProcessGroupStatus(cluster.Status.ProcessGroups, true).Map()
 	desiredCountStruct, err := cluster.GetProcessCountsWithDefaults()
 	if err != nil {
 		return false, err
 	}
 	desiredCounts := desiredCountStruct.Map()
 
+	adminClient, err := r.AdminClientProvider(cluster, r)
+	if err != nil {
+		return false, err
+	}
+	status, err := adminClient.GetStatus()
+	if err != nil {
+		return false, err
+	}
+	localityMap := make(map[string]localityInfo)
+	for _, process := range status.Cluster.Processes {
+		id := process.Locality[FDBLocalityInstanceIDKey]
+		localityMap[id] = localityInfo{ID: id, Address: process.Address, LocalityData: process.Locality}
+	}
+
 	for _, processClass := range fdbtypes.ProcessClasses {
 		desiredCount := desiredCounts[processClass]
 
 		removedCount := currentCounts[processClass] - desiredCount
 
+		processClassLocality := make([]localityInfo, 0, currentCounts[processClass])
+
 		for _, processGroup := range cluster.Status.ProcessGroups {
-			if processGroup.ProcessClass == processClass && processGroup.Remove {
-				removedCount--
+			if processGroup.ProcessClass == processClass {
+				if processGroup.Remove {
+					if processGroup.ProcessClass == processClass && processGroup.Remove {
```

Comment:
```suggestion
					if processGroup.Remove {
```

the expression must be true otherwise we wouldn't be here.

Reply:
I think we can drop the complete if statement in this line